great_expectations/validation_operators/validation_operators.py

```diff
@@ -235,6 +254,13 @@ def __init__(
                     class_name=config["class_name"],
                 )
             self.actions[action_config["name"]] = new_action
+        if notify_before_store and self._using_cloud_context:
+            logger.warning(
```

Comment:
Should we raise a real `warning` here?
It will look more or less the same to a user but it would allow us to optional turn this into an exception when debugging or testing.
```python
import warnings

...

warnings.warn(f"The checkpoints action_list configuration has a notification, {notify_before_store}", category=UserWarning)
```
```